torchvision/datasets/utils.py

```diff
@@ -224,15 +223,25 @@ def download_file_from_google_drive(file_id: str, root: str, filename: Optional[
             params = {'id': file_id, 'confirm': token}
             response = session.get(url, params=params, stream=True)
 
-        if _quota_exceeded(response):
+        # Ideally, one would use response.status_code to check for quota limits, but google drive is not consistent
+        # with their own API, refer https://github.com/pytorch/vision/issues/2992#issuecomment-730614517.
+        # Should this be fixed at some place in future, one could refactor the following to no longer rely on decoding
+        # the first_chunk of the payload
+        response_content_generator = response.iter_content(32768)
+        first_chunk = None
+        while not first_chunk:  # filter out keep-alive new chunks
+            first_chunk = next(response_content_generator)
+
+        if _quota_exceeded(first_chunk):
```

Comment:
I know this doesn't come from this PR but I would suggest to remove the `_quota_exceeded` function and just inline it here. We'll be able to use `raise RuntimeError(msg) from decode_error` which will provide cleaner tracebacks.

Reply:
While I can surely inline the function I don't get what you mean with `raise from UnicodeDecodeError`.
We use the `UnicodeDecodeError` as the passing condition (i.e. `UnicodeDecodeError` will only happen if we have a **valid payload**).

Did I misunderstand you here ? Please clarify (sorry I am new to Exception Chaining).

Reply:
oh you're right, it wouldn't make sense then!